prediction_market_agent/agents/social_media_agent/social_media/twitter_handler.py

```diff
@@ -1,47 +1,80 @@
             keys.twitter_access_token_secret.get_secret_value(),
         )
 
+        self.llm = ChatOpenAI(
+            temperature=0,
+            model=model,
+            api_key=APIKeys().openai_api_key.get_secret_value(),
+        )
+
+    def make_tweet_more_concise(self, tweet: str) -> str:
+        system_template = f"Make this tweet more concise while keeping an analytical tone. You are forbidden of using more than {POST_MAX_LENGTH} characters."
+        prompt_template = ChatPromptTemplate.from_messages(
+            [("system", system_template), ("user", "{text}")]
+        )
+
+        chain = prompt_template | self.llm | StrOutputParser()
+        result = chain.invoke({"text": tweet})
+        return result
+
     @staticmethod
     def does_post_length_exceed_max_length(tweet: str) -> bool:
         return len(tweet) > POST_MAX_LENGTH
 
     def post(self, text: str, reasoning_reply_tweet: str) -> None:
+        quote_tweet_id = self.post_else_retry_with_summarization(text)
+        self.post_else_retry_with_summarization(
+            reasoning_reply_tweet, quote_tweet_id=quote_tweet_id
+        )
+
+    def post_else_retry_with_summarization(
+        self, text: str, quote_tweet_id: str | None = None
+    ) -> str | None:
+        """
+        Posts the provided text on Twitter and retries with a summarized version if the text exceeds the maximum
+        length. If the summarized text also exceeds the maximum length, a warning is logged and the function returns
+        without posting.
+        """
         if self.does_post_length_exceed_max_length(text):
-            logger.warning(
-                f"Tweet too long. Length: {len(text)}, max length: {POST_MAX_LENGTH}"
-            )
-            return
-        first_tweet = self.client.create_tweet(text=text)
-        logger.debug(f"Posted tweet {text} - {first_tweet}")
-
-        if self.does_post_length_exceed_max_length(reasoning_reply_tweet):
-            logger.warning(
-                f"Retweet too long. Length: {len(reasoning_reply_tweet)}, max length: {POST_MAX_LENGTH}"
-            )
-            return
-
-        reply_tweet = self.client.create_tweet(
-            text=reasoning_reply_tweet, quote_tweet_id=first_tweet.data["id"]
+            if self.does_post_length_exceed_max_length(
+                text := self.make_tweet_more_concise(text)
+            ):
+                logger.warning(
+                    f"Tweet too long. Length: {len(text)}, max length: {POST_MAX_LENGTH}"
+                )
+                return None
+        posted_tweet = self.client.create_tweet(
+            text=text, quote_tweet_id=quote_tweet_id
         )
-        logger.debug(f"Posted quote tweet {reasoning_reply_tweet} - {reply_tweet}")
+        logger.debug(f"Tweeted {text} - {posted_tweet}")
+        try:
+            return str(posted_tweet.data["id"])
+        except Exception as e:
+            logger.debug("Could not extract tweet ID. Exception: ", e)
```

Comment:
```suggestion
            logger.exception("Could not extract tweet ID. Exception: ", e)
```

Debug logs aren't shown in either GCP Logs or Kube , would go unnoticed.